src/Integration.Vsix/ErrorList/IssuesSnapshot.cs

```diff
@@ -27,14 +27,35 @@
 using SonarLint.VisualStudio.Core;
 using SonarLint.VisualStudio.Core.Analysis;
 using SonarLint.VisualStudio.IssueVisualization.Helpers;
+using SonarLint.VisualStudio.IssueVisualization.Models;
 using SonarLint.VisualStudio.IssueVisualization.TableControls;
 
 namespace SonarLint.VisualStudio.Integration.Vsix
 {
     internal interface IIssuesSnapshot : ITableEntriesSnapshot
     {
+        /// <summary>
+        /// Logical identifier for a set of issues produced by an analysis run.
+        /// Every snapshot that has the same analysis id describes the same set of
+        /// issues in the same order. The implementation of "IndexOf" depends on this.
+        /// </summary>
         Guid AnalysisRunId { get; }
+
+        /// <summary>
+        /// The list of issues returned by the analyzer run
+        /// </summary>
         IEnumerable<IssueMarker> IssueMarkers { get; }
+
+        /// <summary>
+        /// Returns the set of files for which this snapshot contains location information (primary and secondary)
+        /// </summary>
+        IEnumerable<string> FilesInSnapshot { get; }
+
+        /// <summary>
+        /// Returns all of the location visualizations (primary and secondary) that are in the specified file
+        /// </summary>
+        /// <returns></returns>
+        IEnumerable<IAnalysisIssueLocationVisualization> GetLocationsVizsForFile(string filePath);
```

Comment:
Added two new members that will be consumed by the `IIssueLocationStore` implementation (to be added in the next PR).
The tagger will ask the `IIssueLocationStore` for all issue locations relating to a particular file; the `IIssueLocationStore` will in turn loop round all of the snapshots, calling these methods to get the relevant files.